lib/pbench/agent/tool_meister.py

```diff
@@ -852,20 +806,26 @@ def __enter__(self):
                     name,
                     tool_opts,
                     pbench_install_dir=self.pbench_install_dir,
-                    tool_dir=self._tool_dir,
                     logger=self.logger,
                 )
                 # FIXME - consider running these in parallel.
-                tool_installs[name] = tool.install()
-            except Exception:
+                res = tool.install()
+            except (Warning, RuntimeError, OSError):
                 self.logger.exception("Failed to run tool %s install check", name)
-                tool_installs[name] = (-42, "internal-error")
-        self._failed_tools = {}
-        for name, res in tool_installs.items():
+                res = (-42, "internal-error")
+            # Record the result of the tool installation check so it can be
+            # reported back to the Tool Data Sink.
+            tool_installs[name] = res
```

Comment:
Perhaps in a separate PR?

Reply:
Given that you've got approvals from me and Dave, I'm not going to hold up this PR for this change...but it would be good to see it show up in a subsequent PR.